std/typecons.d

```diff
@@ -792,10 +852,36 @@ template Tuple(Specs...)
          * Returns:
          *     The string representation of this `Tuple`.
          */
-        string toString()()
+        string toString()() const
         {
-            import std.conv : to;
-            return this.to!string;
+            import std.array : appender;
+            auto app = appender!string();
+            this.toString((const(char)[] chunk) => app ~= chunk);
+            return app.data;
+        }
+
+        ///
+        unittest
```

Comment:
Unit tests are used to verify that certain instantiations work with certain attributes:
```d
void foo(T)(T t);
@safe unittest
{
    foo(42); // if this compiles, we've tested that this instantiation is @safe
}
```
A unit test without any attributes doesn't test any.